simulation/agents.py

```diff
@@ -129,7 +186,52 @@ def decide_renovation_scope(self) -> None:
         )
         self.renovate_insulation = self.true_with_probability(PROBA_INSULATION_UPDATE)
 
+    def get_upgradable_insulation_elements(self) -> Set[Element]:
+
+        measures_and_grades = zip(
+            [Element.WALLS, Element.ROOF, Element.GLAZING],
+            [
+                self.walls_energy_efficiency,
+                self.roof_energy_efficiency,
+                self.windows_energy_efficiency,
+            ],
+        )
+
+        MAX_ENERGY_EFFICIENCY_SCORE = 5
+        return set(
+            [
+                measure
+                for measure, grade in measures_and_grades
+                if grade < MAX_ENERGY_EFFICIENCY_SCORE
+            ]
+        )
```

Comment:
Minor syntax suggestion: like list comprehnsions, you can do set comprehensions, e.g. `{measure for measure, grade in measures_and_grades if grade < MAX_ENERGY_EFFICIENCY_SCORE}`.